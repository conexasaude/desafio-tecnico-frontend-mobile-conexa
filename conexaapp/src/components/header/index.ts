import {Button} from './button/view';
import {Greetings} from './greetings/view';
import {HStack} from './horizontal-stack/view';
import {Root} from './root/view';
import {Title} from './title/view';
import {VStack} from './vertical-stack/view';

export const Header = {
  Root: Root,
  Button: Button,
  Greetings: Greetings,
  VStack: VStack,
  HStack: HStack,
  Title: Title,
};
