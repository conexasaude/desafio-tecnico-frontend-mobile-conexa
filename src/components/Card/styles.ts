import styled, { css } from '@emotion/native'

export const Container = styled.View`
  border-radius: 5px;
  width: 100%;
  padding: 16px;

  ${({ theme }) => css`
    background-color: ${theme.COLORS.WHITE};
  `};
`
