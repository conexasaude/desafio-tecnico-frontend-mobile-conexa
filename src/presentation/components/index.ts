export * from './buttons';
export * from './heading';
export * from './icons';
export * from './inputs';
