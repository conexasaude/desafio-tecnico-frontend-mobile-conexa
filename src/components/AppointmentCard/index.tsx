import { GestureResponderEvent, Pressable } from 'react-native';
import { Container, Pacient, AppointmentDate, Field, Title, Header } from './styles';
import { MaterialCommunityIcons } from '@expo/vector-icons'; 
import formatsDate from '../../utils/formatsDate';

interface AppointmentCardProps {
    id: string
    onPress: (event: GestureResponderEvent) => void
    pacient: string
    date: string
    observation: string
}

export default function AppointmentCard({ id, date, observation, onPress, pacient }: AppointmentCardProps) {

  return (
    <Container id={id} onPress={onPress}>
        <Header>
            <Title>Consulta #{id}</Title>
            <AppointmentDate>{formatsDate(date, 'day-month-year')}</AppointmentDate>
        </Header>
        <Field>
            <MaterialCommunityIcons name="account" size={20} color="#FFF" />
            <Pacient>{pacient}</Pacient>
        </Field>
        <Field>
            <MaterialCommunityIcons name="clock" size={20} color="#FFF" />
            <AppointmentDate>{formatsDate(date, 'time')}</AppointmentDate>
        </Field>
    </Container>
  );
}
