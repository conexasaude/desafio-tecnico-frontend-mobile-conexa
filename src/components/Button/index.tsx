import { ButtonContainer, ButtonText } from './styles';

interface ButtonProps {
  onPress: () => Promise<void>
  label: string
  disabled?: boolean
  testID?: string
}

export default function Login({ label, onPress, disabled, testID }: ButtonProps) {

  return (
    <ButtonContainer testID={testID} onPress={onPress} disabled={disabled}>
      <ButtonText>{label}</ButtonText>
    </ButtonContainer>
  );
}