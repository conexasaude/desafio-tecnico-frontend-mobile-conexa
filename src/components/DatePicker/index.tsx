import { GestureResponderEvent, Pressable, Text } from 'react-native';
import { Label, Placeholder, Value, ValueContainer } from './styles';

interface DatePickerProps {
    label: string
    value: Date | null
    onPress: ((event: GestureResponderEvent) => void) | null | undefined
}

export default function DatePicker({ label, value, onPress }: DatePickerProps) {
    function valueFormat(date: Date) {
        const formattedDate = new Intl.DateTimeFormat('pt-BR', {
			year: 'numeric',
			month: '2-digit',
			day: '2-digit',
			hour: '2-digit',
			minute: '2-digit',
		}).format(date);

        return formattedDate
    }
    return (
        <Pressable onPress={onPress}>
            <Label>
                {label}
                <Text style={{ color: 'red' }}> *</Text>
            </Label>
            
            <ValueContainer>
                {value ? <Value>{valueFormat(value)}</Value> : <Placeholder>Selecione uma data</Placeholder> }
            </ValueContainer>
        </Pressable>
    )
}