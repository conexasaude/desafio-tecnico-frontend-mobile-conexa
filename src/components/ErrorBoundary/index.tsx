import React from 'react';
import { useTheme } from '@emotion/react';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import * as Styled from './styles';

interface ErrorBoundaryProps {
  tryAgain: () => void;
}

export function ErrorBoundary({ tryAgain }: ErrorBoundaryProps) {
  const { colors } = useTheme();

  return (
    <Styled.Container testID="error-boundary">
      <Icon
        name="alert-circle-outline"
        color={colors.status.warning}
        size={64}
      />
      <Styled.Text>
        {'Desculpe, ocorreu um problema. Por favor, tente novamente.'}
      </Styled.Text>

      <Styled.ButtonTryAgain onPress={tryAgain} title="Tentar Novamente" />
    </Styled.Container>
  );
}
