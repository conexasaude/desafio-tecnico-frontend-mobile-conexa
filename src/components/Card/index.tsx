import React from 'react';
import * as Styled from './styles';
import { TouchableOpacityProps } from 'react-native';
import { ButtonProps } from '@/components/Button';

interface CardProps extends TouchableOpacityProps {
  title: string;
  description: string;
  button: ButtonProps;
}

export function Card({ title, description, button, ...rest }: CardProps) {
  return (
    <Styled.Wrapper
      testID="card"
      activeOpacity={0.8}
      onPress={button.onPress}
      {...rest}>
      <Styled.Row>
        <Styled.Column>
          <Styled.Title numberOfLines={1}>{title}</Styled.Title>
          <Styled.Description numberOfLines={2}>
            {description}
          </Styled.Description>
        </Styled.Column>

        <Styled.SeeMore
          testID="card-button"
          variant="outlined"
          activeOpacity={0.4}
          {...button}
        />
      </Styled.Row>
    </Styled.Wrapper>
  );
}
