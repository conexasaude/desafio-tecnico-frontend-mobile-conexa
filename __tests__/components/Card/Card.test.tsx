import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/lib/functions/Testing';
import { Card } from '@/components/Card';
import { fireEvent } from '@testing-library/react-native';

describe('Card', () => {
  test('should render correctly', () => {
    renderWithThemeProvider(
      <Card
        title="title"
        description="description"
        button={{ title: 'ver mais' }}
      />,
    );
  });

  test('should matches snapshot', () => {
    matchSnapshotWithProvider(
      <Card
        title="title"
        description="description"
        button={{ title: 'ver mais' }}
      />,
    );
  });

  test('should render title description and button', () => {
    const { getByText, getByTestId } = renderWithThemeProvider(
      <Card
        title="title"
        description="description"
        button={{ title: 'ver mais' }}
      />,
    );

    expect(getByText('title')).toBeDefined();
    expect(getByText('description')).toBeDefined();
    expect(getByTestId('card-button')).toBeDefined();
  });

  test('should called onPress button when press card', () => {
    const press = jest.fn();

    const { getByTestId } = renderWithThemeProvider(
      <Card
        title="title"
        description="description"
        button={{ title: 'ver mais', onPress: press }}
      />,
    );

    fireEvent.press(getByTestId('card'));
    expect(press).toHaveBeenCalledTimes(1);

    fireEvent.press(getByTestId('card-button'));
    expect(press).toHaveBeenCalledTimes(2);
  });
});
