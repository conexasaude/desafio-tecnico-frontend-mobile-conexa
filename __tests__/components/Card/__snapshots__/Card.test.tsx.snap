// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#fff",
      "borderRadius": 8,
      "elevation": 2,
      "marginBottom": 16,
      "opacity": 1,
      "padding": 12,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 1.5,
      "width": "100%",
    }
  }
  testID="card"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
            "height": 52,
            "justifyContent": "space-around",
            "marginRight": 4,
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "color": "#7F7F7F",
              "fontFamily": "Nunito-SemiBold",
              "fontSize": 16,
            },
            undefined,
          ]
        }
      >
        title
      </Text>
      <Text
        numberOfLines={2}
        style={
          [
            {
              "color": "#999999",
              "fontFamily": "Nunito-Regular",
              "fontSize": 14,
            },
            undefined,
          ]
        }
      >
        description
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderColor": "#2E50D4",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 8,
          "width": "auto",
        }
      }
      testID="card-button"
    >
      <Text
        style={
          [
            {
              "color": "#2E50D4",
              "fontFamily": "Nunito-Bold",
              "fontSize": 14,
              "textAlign": "center",
            },
            undefined,
          ]
        }
        variant="outlined"
      >
        ver mais
      </Text>
    </View>
  </View>
</View>
`;
